app/auth/views.py

```diff
@@ -0,0 +1,157 @@
+from . import auth_blueprint
+from flask.views import MethodView
+from flask import make_response, request, jsonify
+from app.models import User, BlackListToken
+from flasgger import swag_from
+
+
+def get_authenticated_user(request):
+    auth_header = request.headers.get('Authorization')
+    if not auth_header:
+        return None
+    try:
+        access_token = auth_header
+    except (IndexError, ValueError):
+        return 'Authorization header is in wrong format.'
+    if not access_token:
+        return None
+    else:
+        user_id = User.decode_token(access_token)
+        if not isinstance(user_id, str):
+            # user is authenticated so get the user
+            user = User.query.get(user_id)
+            return user
+        elif user_id == 'You are already logged out':
+            return user_id
+        else:
+            return None
+
+
```

Comment:
Place line 8-27 in a different file. You can look into how to make a decorator. It'll help you avoid repetition every time you access endpoints.